tests/js/set-tests.js

```diff
@@ -0,0 +1,392 @@
+        TestCase.assertEqual(false, js_team.Names.has("Jason"), "Jason shouldn't be in the JS team");
+
+        // add one JS team member, delete another one
+        realm.write(() => {
+            js_team.Names.add("Kræn").delete("Kenneth");
+        });
+        teams = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, teams.length, "There should be only one JS team");
+        js_team = teams[0];
+        TestCase.assertEqual(3, js_team.Names.length, "There should be 3 people in the JS team");
+        TestCase.assertEqual(false, js_team.Names.has("Kenneth"), "Kenneth shouldn't be in the JS team");
+
+        // create another team with two people
+        realm.write(() => {
+            realm.create(people_schema.name, {
+                Names: ["Christian", "Claus"]
+            });
+        });
+        teams = realm.objects(people_schema.name);
+        TestCase.assertEqual(2, teams.length, "There should be two teams");
+        js_team = teams[0];
+        let java_team = teams[1];
+        TestCase.assertEqual(3, js_team.Names.length, "There should be 3 people in the JS team");
+        TestCase.assertEqual(2, java_team.Names.length, "There should be 2 people in the Java team");
+
+        TestCase.assertEqual(false, java_team.Names.has("Kenneth"), "Kenneth shouldn't be in the Java team");
+        TestCase.assertEqual(true, java_team.Names.has("Claus"), "Claus should be in the Java team");
+
+        //
+        // Set.clear() functionality
+        //
+
+        // remove everyone from the JS team
+        realm.write(() => {
+            js_team.Names.clear();
+        });
+
+        teams = realm.objects(people_schema.name);
+        TestCase.assertEqual(2, teams.length, "There should be two teams");
+        js_team = teams[0];
+        java_team = teams[1];
+        TestCase.assertEqual(0, js_team.Names.length, "There should be no people in the JS team");
+        TestCase.assertEqual(2, java_team.Names.length, "There should be 2 people in the Java team");
+
+        realm.close();
+    },
+
+    // TODO:  Move Sync tests to separate file
+    async testSetSyncedNonRequired() {
+        // test that we can create a synced realm with a Set
+        // that isn't required
+        if (!global.enableSyncTests) return;
+
+        const schema = {
+            name: "SyncedSetInt",
+            primaryKey: "_id",
+            properties: {
+                _id: "int",
+                intSet: "int<>",
+            }
+        }
+
+        const appConfig = AppConfig.integrationAppConfig;
+        const app = new Realm.App(appConfig);
+        const credentials = Realm.Credentials.anonymous();
+
+        const user = await app.logIn(credentials);
+        const config = {
+            sync: {
+                user,
+                partitionValue: "_id",
+                _sessionStopPolicy: "immediately", // Make it safe to delete files after realm.close()
```

Comment:
Shouldn't we register a logger and ensure no errors are emitted? Since a sync error wouldn't result in an error being thrown when we call our user-facing API.

Reply:
Scrap that, we should probably upload changes, delete the client-side realm and re-download to verify the round-trip.